File: src/main/java/com/github/ccguyka/showupdates/filter/DependencyManagementFilter.java
package com.github.ccguyka.showupdates.filter;

import static java.util.stream.Collectors.toList;

import java.util.List;
import java.util.Objects;

import org.apache.maven.model.Dependency;
import org.apache.maven.plugin.logging.Log;
import org.apache.maven.project.MavenProject;

import com.github.ccguyka.showupdates.filter.pom.Project;

public class DependencyManagementFilter extends DependencyFilter {

    public DependencyManagementFilter(MavenProject project, Log log) {
        super(project, log);
    }

    @Override
    protected List<Dependency> filter(final List<Dependency> dependencies, Project pom) {
        return dependencies.stream()
                .filter(dependency -> pom.getDependencyManagement().getDependencies()
                        .stream()
                        .anyMatch(dep -> Objects.equals(dep.getArtifactId(), dependency.getArtifactId())
                                && Objects.equals(dep.getGroupId(), dependency.getGroupId())))
                .collect(toList());
    }
}

File: src/main/kotlin/com/github/ccguyka/showupdates/filter/DependencyManagementFilter.kt
package com.github.ccguyka.showupdates.filter

import com.github.ccguyka.showupdates.filter.pom.Project
import org.apache.maven.model.Dependency
import org.apache.maven.plugin.logging.Log
import org.apache.maven.project.MavenProject
import java.util.stream.Collectors

class DependencyManagementFilter(project: MavenProject?, log: Log?) : DependencyFilter(project!!, log!!) {

    override fun filter(dependencies: List<Dependency>, pom: Project): List<Dependency> {
        return dependencies.stream()
                .filter { dependency: Dependency ->
                    pom.dependencyManagement.dependencies
                            .stream()
                            .anyMatch { dep: com.github.ccguyka.showupdates.filter.pom.Dependency ->
                                (dep.artifactId == dependency.artifactId
                                        && dep.groupId == dependency.groupId)
                            }
                }
                .collect(Collectors.toList())
    }
}
